libyul/optimiser/SyntacticalEquality.h

```diff
@@ -58,6 +58,8 @@ class SyntacticallyEqual
 	bool statementEqual(Break const&, Break const&) { return true; }
 	bool statementEqual(Continue const&, Continue const&) { return true; }
 	bool statementEqual(Block const& _lhs, Block const& _rhs);
+	bool statementEqual(Comment const&, Comment const&) { return true; }
```

Comment:
Are you sure this is enough? Visits between mismatching nodes will always return ``false``, so

```
{
  // comment
  mstore(0,1)
}
```
and
```
{
  mstore(0,1)
}
```
will now still *not* be syntactically equal...

To avoid that I think ``statementEqual(Block const&, Block const&)`` has to be reworked to skip over all comments instead...